### PR TITLE
bot: Retry `get_program_accounts` if it times out

### PR DESCRIPTION
**Pull request by joncinque**
#### Problem

The bot dies if it reaches a timeout on `get_program_accounts` when fetching all stake accounts. This is an extremely heavy operation, so it makes sense that it times out once in awhile.

#### Solution

As a first step, just wait and retry if there's a timeout.

This re-enables the stake_pool_v0 test, which was succeeding in #465, but was ignored.

### CODE REVIEW

File: bot/src/stake_pool.rs [hunk 1]
```diff
@@ -882,7 +882,10 @@ mod test {
     };
 
     fn num_stake_accounts(rpc_client: &RpcClient, authority: Pubkey) -> usize {
```
Comment (t-nelson) [inline bot/src/stake_pool.rs:884]: aside: wtf?  can't we just resolve this from the big grab at the start?

File: bot/src/rpc_client_utils.rs [hunk 2]
```diff
@@ -338,31 +340,61 @@ pub fn get_vote_account_info(
...
+        }
+        let maybe_accounts =
+            rpc_client.get_program_accounts_with_config(&stake::program::id(), config.clone());
+
+        match maybe_accounts {
+            Err(ClientError {
+                kind: ClientErrorKind::Reqwest(err),
+                ..
+            }) if reqwest::Error::is_timeout(&err) => {
+                retries = retries.saturating_sub(1);
```
Comment (t-nelson) [inline bot/src/rpc_client_utils.rs:377]: this is gonna loop forever if we hit some other http error.  can switch to a for-loop+break and move the `if retries == 0` block to immediately after the loop

File: bot/src/stake_pool.rs (hunk 1 above)
Comment (joncinque) [reply bot/src/stake_pool.rs:884]: No worries, this is just for the tests!

File: bot/src/rpc_client_utils.rs (hunk 2 above)
Comment (joncinque) [reply bot/src/rpc_client_utils.rs:377]: Shouldn't it return the error? That's the intent with the catch-all underneath it, unless I did it wrong

Comment (joncinque) [reply bot/src/rpc_client_utils.rs:377]: I tested this just to be sure, and this will properly fall through on any other error

Comment (t-nelson) [reply bot/src/rpc_client_utils.rs:377]: oh haha, dammit i had a pending comment from weeks ago here that i apparently never submitted the approval review for :facepalm: 

got confused by the hiding `Ok(_) | Err(_)` in `_ => { ... }`

**Review by t-nelson — APPROVED** [review]
:shipit:

**Comment by joncinque** [comment]
@tigarcia I can't merge, so could you do the honors?

**Pull request merged**